File: takasugi/Python/NLP-100-09.py
#
# -*- coding: utf-8 -*-

import random


def scramble(str):
    if len(str) < 5:
        return str
    s_list = list(str)

    head_chr = s_list.pop(0)
    tail_chr = s_list.pop()

    random.shuffle(s_list)

    s_list.insert(0, head_chr)
    s_list.append(tail_chr)

    return ''.join(s_list)


a = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."

a_list = a.split(' ')
b_list = map(lambda n: scramble(n), a_list)

print "Before:"
print a
print "After:"
print ' '.join(b_list)

File: takasugi/Ruby/NLP-100-09.rb
#

def randamize_ary(str)
  str.shuffle!
end

def randamize(word)
  return word if word.length < 5

  ary = word.chars

  head = ary.shift
  tail = ary.pop

  body = randamize_ary(ary)

  out = body.unshift(head)
  out << tail

  out.join
end

puts randamize('123')

puts randamize('a123b')

sentence = "I couldn't believe that I could actually understand what I was reading : the phenomenal power of the human mind ."

n_sntc = sentence.split(/ /)
out = n_sntc.map{|w| randamize(w)}

puts out.join(' ')
